openupgrade_scripts/scripts/stock/18.0.1.1/pre-migration.py

```diff
@@ -48,6 +52,30 @@ def fill_stock_move_location_dest_id(env):
         WHERE sm2.id = sm.id
         """,
     )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE stock_move sm2
+        SET location_final_id = sm.location_final_id
+        FROM stock_move sm
+        JOIN stock_move_move_rel rel ON rel.move_dest_id = sm.id
```

Comment:
this should be recursive I think, you can have chains of push moves of arbitrary length

Reply:
recursion done!